src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/ListView/ListView_DataContext_Reference.xaml.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices.WindowsRuntime;
+using Uno.UI.Samples.Controls;
+using Windows.Foundation;
+using Windows.Foundation.Collections;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Navigation;
+using System.Collections.ObjectModel;
+using System.Windows.Input;
+using System.ComponentModel;
+
+// The User Control item template is documented at https://go.microsoft.com/fwlink/?LinkId=234236
+namespace UITests.Shared.Windows_UI_Xaml_Controls.ListView
+{
+	[SampleControlInfo("ListView", "ListView_DataContext_Reference", viewModelType: typeof(DataContextReferenceViewModel), isManualTest: true)]
+	public sealed partial class ListView_DataContext_Reference : UserControl
+	{
+		public ListView_DataContext_Reference()
+		{
+			this.InitializeComponent();
+		}
+	}
+
+	public class DataContextReferenceViewModel
+	{
+		public ObservableCollection<Item> Items { get; } =
+			new ObservableCollection<Item>(Enumerable.Range(0, 10).Select(x => new Item(x)));
+
+		private ICommand _command;
+		public ICommand Command
+		{
+			get
+			{
+				_command ??= new Command<Item>(ExecuteCommand);
+				return _command;
+			}
+		}
+
+		private void ExecuteCommand(Item item)
+		{
+			if (item is null)
+			{
+				Console.WriteLine("Item was null");
+				return;
+			}
+
+			var newItem = item with { Name = $"Selected item {item.Index}" };
+
+			Items.Remove(item);
+			Items.Insert(newItem.Index, newItem);
```

Comment:
this should be verifiable under runtime test, no?

Reply:
Still trying to figure it out. Once we Remove then Insert a new Item, we need to access the nth-ListViewitem to validate if the `GetParent()` method returns other than `null.`

Reply:
You'll likely need to scroll to the required position. That should be working properly.

Reply:
Added Unit test validating if the container's parent for the added item is not null. This is the behaviour I tested manually when finding the fix.